go-controller/pkg/ovn/pods.go

```diff
@@ -192,7 +192,9 @@ func (oc *Controller) deleteLogicalPort(pod *kapi.Pod) (err error) {
 		}
 	}
 	podNsName := ktypes.NamespacedName{Namespace: pod.Namespace, Name: pod.Name}
-	oc.deleteGWRoutesForPod(podNsName, portInfo.ips)
+	if err := oc.deleteGWRoutesForPod(podNsName, portInfo.ips); err != nil {
+		return err
```

Comment:
part of this function (extSwitchPrefix) will error if the node doesn't exist anymore, which would make retrying later useless. If the node is gone, then we should have deleted the GR (and all ecmp routes) and LS, but not the ovn_cluster_router obviously. So we still need to ensure we remove any hybridRoutePolicyForPod there.